document_ai_warehouse/batch-ingestion-docai-wh/load_docs.py

```diff
@@ -0,0 +1,493 @@
+                    if (
+                        schema
+                        and len(keys) != 0
+                        and len(schema.property_definitions) == 0
+                        and options
+                    ):
+                        create_new_schema = True
+                else:
+                    create_new_schema = True
+
+            if create_new_schema:
+                schema_path = create_mapping_schema(schema_name, keys, options)
+                new_schema_id = create_document_schema(schema_path, True)
+                if document_schema_id != new_schema_id:
+                    created_schemas.add(new_schema_id)
+                    document_schemas[schema_name] = new_schema_id
+                    document_schema_id = new_schema_id
+
+            (parent_id, reference_id) = files_to_parse[f_uri]
+            schema = dw_utils.get_document_schema(document_schema_id)
+
+            metadata_properties = get_metadata_properties(keys, schema)
+            try:
+                upload_document_gcs(
+                    f_uri,
+                    document_schema_id,
+                    parent_id,
+                    reference_id,
+                    document_ai_output,
+                    metadata_properties,
+                )
+            except Exception as ex:
+                Logger.error(f"Failed to upload {f_uri} - {ex}")
+
+    process_time = time.time() - initial_start_time
+    time_elapsed = round(process_time)
+    document_schema_str = ""
+    if len(created_schemas) > 0:
+        document_schema_str = (
+            f"  - created document schema with id {','.join(list(created_schemas))}"
+        )
+    Logger.info(
+        f"Job Completed in {str(round(time_elapsed / 60))} minute(s): \n"
+        f"{document_schema_str}  \n"
+        f"  - processed gcs files={len(processed_files)} \n"
+        f"  - created dw documents={len(document_id_list)} \n"
+        f"  - processed gcs directories={len(processed_dirs)} \n"
+        f"  - created dw directories={len(created_folders)} \n"
+    )
+
+    if len(error_files) != 0:
+        Logger.info(
+            f"Following files could not be handled (Document page number exceeding limit of 200 pages? {','.join(error_files)}"
+        )
+
+
+def get_type(value: str):
+    if type(value) == bool or str(value) == "":
+        return "text_type_options"  # bool Not Supported
+    if is_date(value):
+        return "date_time_type_options"
+    if is_valid_int(value):
+        return "integer_type_options"
+    if is_valid_float(value):
+        return "float_type_options"
+
+    return "text_type_options"
+
+
+def is_valid_float(string: str):
+    try:
+        float(string)
+        return True
+    except ValueError:
+        return False
+
+
+def is_valid_bool(string: str):
+    return string.lower() in ["true", "false"]
+
+
+def is_valid_int(string: str):
+    return string.isdigit()
```

Comment:
Not sure about this, but I think there are built in ways to handle most of this

Reply:
I have found even a single example. I wish there were something

Reply:
If you could verify that, would appreciate it grately